### ISSUE TITLE
fix: Fixed + ellipsis cut off in FluentOverflow

### ISSUE THREAD

**Issue by JamesNK** [microsoft/fluentui-blazor#2427]
# 🐛 Bug Report

FluentOverflow with the first item with `OverflowItemFixed.Ellipsis` is cut off when there are multiple items afterwards:

![image](https://github.com/user-attachments/assets/62eced2d-5c08-4267-9510-cddc3100becc)

After resizing the column, the problem fixes itself:

![overflow-first-cutoff](https://github.com/user-attachments/assets/4452286f-6875-47c3-8f16-60d4d65c2047)

## 💻 Repro or Code Sample

https://github.com/dotnet/aspire/pull/4976

## 🤔 Expected Behavior

First item is only cut off with an ellipsis if there are no other items displayed and there isn't enough room.

## 😯 Current Behavior

See picture.

## 🌍 Your Environment

* OS & Device: [e.g. MacOS, iOS, Windows, Linux] on [iPhone 7, PC]
* Browser [e.g. Microsoft Edge, Google Chrome, Apple Safari, Mozilla FireFox]
* .NET and Fluent UI Blazor library Version [e.g. 8.0.2 and 4.4.1]

**Comment by dvoituron** [comment]
Are you using `VisibleOnLoad`?

**Comment by JamesNK** [comment]
No.

I tried out `VisibleOnLoad` a few days ago and it didn't work well for me. Sometimes overflows never became visible.

**Comment by dvoituron** [comment]
Ok. I will check and fix that quickly.

**Comment by dvoituron** [comment]
I tried to reproduce this issue in our Demo site, but without success.
```xml
<FluentOverflow Style="width: 100%;">
    <FluentOverflowItem Fixed="OverflowItemFixed.Ellipsis">http://localhost:5049,</FluentOverflowItem>
    <FluentOverflowItem>http://localhost:5050,</FluentOverflowItem>
    <FluentOverflowItem>http://localhost:5051,</FluentOverflowItem>
    <FluentOverflowItem>http://localhost:5052,</FluentOverflowItem>
    <FluentOverflowItem>http://localhost:5053,</FluentOverflowItem>
    <FluentOverflowItem>http://localhost:5054,</FluentOverflowItem>
    <FluentOverflowItem>http://localhost:5055,</FluentOverflowItem>
    <FluentOverflowItem>http://localhost:5055,</FluentOverflowItem>
    <FluentOverflowItem>http://localhost:5056,</FluentOverflowItem>
</FluentOverflow>
```

![image](https://github.com/user-attachments/assets/4519fab1-bfd9-49f7-b199-6eaa4d1327dc)

I will try using the Aspire Repo.

**Comment by dvoituron** [comment]
@JamesNK The problem seems to be related to this `flex` style applied to the **FluentButton**.
Strange, but I think that with `flex' the JavaScript code cannot calculate the initial element widths correctly.

```css
.endpoint-overflow .fluent-overflow-more {
    display: flex;
}
```

I found this solution, replacing the **FluentButton** by a **FluentBadge**.

```xml
<MoreButtonTemplate Context="overflow">
    <FluentBadge role="button" tabindex="0" Appearance="Appearance.Accent" OnClick="() => _popoverVisible = !_popoverVisible" Class="endpoint-button">
        @($"+{overflow.ItemsOverflow.Count()}")
    </FluentBadge>
</MoreButtonTemplate>
```

And using this style to get the same look and feel.

```css
/*
.endpoint-overflow .fluent-overflow-more {
    display: flex;
}
*/

.endpoint-button::part(control) {
    padding: 2px 8px !important;
}
```

**Comment by dvoituron** [comment]
I pushed this PR: https://github.com/dotnet/aspire/pull/5030

**Comment by JamesNK** [comment]
I had an alternative fix that appears to work:

```diff
-.endpoint-overflow .fluent-overflow-more {
-    display: flex;
- }
+.endpoint-overflow .fluent-overflow-more fluent-button {
+    vertical-align: middle;
+ }
```

**Comment by dvoituron** [comment]
Yeah. Perfect and easier. 
The main part is to not use `flex`. 😀

**Comment by JamesNK** [comment]
Another place where I'm doing this appears to have the same or similar problem:

![image](https://github.com/user-attachments/assets/7c945d5e-e3f8-4531-9b35-c11209aa9430)

**Comment by JamesNK** [comment]
I think I fixed it:

```xml
<MoreButtonTemplate Context="overflow">
    <span class="dimension-tag" style="display:inline-block;">
        @($"+{overflow.ItemsOverflow.Count()}")
    </span>
</MoreButtonTemplate>
```

Adding `style="inline-block;"` made it work. I think any inline or flex content in the `MoreButtonTemplate` is a problem.

**Comment by dvoituron** [comment]
Yes. Probably during the initiation process where JS cannot calculate the "real" item width. I will add a note in the `FluentOverflow` documentation.
